### PR TITLE
magento/devdocs#: Layout instructions. Adding examples

### PR DESCRIPTION
**Pull request by DenisSaltanahmedov**
## Purpose of this pull request

This pull request (PR) adds the code examples to the Developer roadmap topic.

## Affected DevDocs pages
-  https://devdocs.magento.com/guides/v2.3/extension-dev-guide/intro/developers_roadmap.html

## Links to Magento source code

-  https://github.com/magento/devdocs/blob/master/src/guides/v2.3/extension-dev-guide/intro/developers_roadmap.md

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.3/extension-dev-guide/intro/developers_roadmap.md [hunk 1]
```diff
@@ -12,15 +12,92 @@ To satisfy the minimum required elements for creating or customizing your Magent
 
 *  [Declare component dependencies]({{ page.baseurl }}/extension-dev-guide/build/composer-integration.html) in `composer.json`.
```
Comment (rogyar) [inline src/guides/v2.3/extension-dev-guide/intro/developers_roadmap.md]: If we extend this info with the example, we may clarify where exactly the dependencies are declared.

```suggestion
*  [Declare component dependencies]({{ page.baseurl }}/extension-dev-guide/build/composer-integration.html) in the `require` and/or `require-dev` section of the `composer.json`.
```

File: src/guides/v2.3/extension-dev-guide/intro/developers_roadmap.md [hunk 2]
```diff
@@ -12,15 +12,92 @@ To satisfy the minimum required elements for creating or customizing your Magent
...
+    ```php
+    <?php
+    use \Magento\Framework\Component\ComponentRegistrar;
+    ComponentRegistrar::register(ComponentRegistrar::MODULE, 'Magento_Config', __DIR__);
+    ```
+
 *  Use these component-specific XML definition files:
    *  Modules: [`module.xml`]({{ page.baseurl }}/extension-dev-guide/build/create_component.html)
+
+       ##### Example: `module.xml` file.
```
Comment (rogyar) [inline src/guides/v2.3/extension-dev-guide/intro/developers_roadmap.md]: I would suggest using relative path here to make it more clear where exactly this file should be placed/edited

```suggestion
       ##### Example: `etc/module.xml` file.
```

This suggestion also applies to all other examples.

**Comment by rogyar** [comment]
Hi @DenisSaltanahmedov. Thank you for your collaboration. Would you provide more information regarding why do we need this information/change next time, please? It would be very helpful upon processing the pull request. 

Thank you!

**Comment by DenisSaltanahmedov** [comment]
Hi @rogyar. Thank you for your help, I have fixed the recommendations.

**Comment by dobooth** [comment]
Thanks for the contribution. I think I need to close this PR. This page is supposed to be a high-level overview of the extension process. High-level page shouldn't have lots of code samples: Those come later in the guide. If anything, I would rather the major extension steps be numbers and made a bit more clear, with links to these steps.

**Pull request closed without merging**